src/activity_handlers.cpp

```diff
@@ -1274,6 +1275,38 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
     resume_for_multi_activities( *p );
 }
 
+void activity_handlers::milk_finish( player_activity *act, player *p )
+{
+    if( act->coords.empty() ) {
+        debugmsg( "milking activity with no position of monster stored" );
+        return;
+    }
+    const tripoint source_pos = g->m.getlocal( act->coords.at( 0 ) );
+    Creature *c = g->critter_at( source_pos );
+    monster *source_mon = dynamic_cast<monster *>( c );
+    if( source_mon == nullptr ) {
+        debugmsg( "could not find source creature for liquid transfer" );
+        return;
+    }
+    const auto milked_item = source_mon->ammo.find( "milk_raw" );
+    if( milked_item == source_mon->ammo.end() ) {
+        debugmsg( "animal has no milk ammo type" );
+        return;
+    }
```

Comment:
if wouldn't hurt to also check for `milked_item->second` being greater than zero.